source/common/router/vhds.cc

```diff
@@ -37,16 +37,16 @@ VhdsSubscription::VhdsSubscription(
       route_config_providers_(route_config_providers) {
   const auto& config_source = config_update_info_->routeConfiguration()
                                   .vhds()
-                                  .config_source()
-                                  .api_config_source()
-                                  .api_type();
-  if (config_source != envoy::config::core::v3::ApiConfigSource::DELTA_GRPC) {
+                                  .config_source();
+  if (config_source.has_api_config_source() &&
```

Comment:
`config_source` field (https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route.proto#config-route-v3-vhds) is required in VHDS config, this check is unnecessary.

Reply:
I think the idea is the config source would point at ADS rather than an API config source. This only makes sense though if the underlying ADS stream is delta gRPC. Should we have a check for that as well?

Also needs tests, thanks.
/wait